### PR TITLE
Depublishing some app layer methods

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Depublishing some methods that are not used outside the app layer

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/file.go [hunk 1]
```diff
@@ -733,7 +735,8 @@ func (t *UploadFileTask) init(a *App) {
 // upload, returning a rejection error. In this case FileInfo would have
 // contained the last "good" FileInfo before the execution of that plugin.
 func (a *App) UploadFileX(c request.CTX, channelID, name string, input io.Reader,
-	opts ...func(*UploadFileTask)) (*model.FileInfo, *model.AppError) {
+	opts ...func(*UploadFileTask),
```
Comment (hanzei) [inline server/channels/app/file.go:738]: Nit: It's easier to read if `opts` is part of the same line as the other arguments.

File: server/channels/app/options.go [hunk 2]
```diff
@@ -110,8 +110,10 @@ func SkipPostInitialization() Option {
 	}
 }
 
-type AppOption func(a *App)
-type AppOptionCreator func() []AppOption
+type (
+	AppOption        func(a *App)
+	AppOptionCreator func() []AppOption
+)
```
Comment (hanzei) [inline server/channels/app/options.go:116]: 1/5 is quite uncommon to use `type` blocks. The pervious pattern is fine

File: server/channels/app/server.go [hunk 3]
```diff
@@ -1700,15 +1695,7 @@ func (s *Server) GetMetrics() einterfaces.MetricsInterface {
 	return s.platform.Metrics()
 }
 
-// SetRemoteClusterService sets the `RemoteClusterService` to be used by the server.
-// For testing only.
-func (s *Server) SetRemoteClusterService(remoteClusterService remotecluster.RemoteClusterServiceIFace) {
-	s.serviceMux.Lock()
-	defer s.serviceMux.Unlock()
-	s.remoteClusterService = remoteClusterService
-}
```
Comment (hanzei) [inline server/channels/app/server.go:1709]: Are you sure this is no longer needed? Are plugins using it?

**Review by hanzei — APPROVED** [review]
LGTM, a few non blocking comments below

File: server/channels/app/server.go (hunk 3 above)
Comment (jespino) [reply server/channels/app/server.go:1709]: Plugins shouldn't have access to anything inside the application without explicitly exposing it, so if the compilation works, it should be working.

File: server/channels/app/options.go (hunk 2 above)
Comment (jespino) [reply server/channels/app/options.go:116]: This is reformated automatically by my editor, I don't know exactly why, let me double check.

File: server/channels/app/file.go (hunk 1 above)
Comment (jespino) [reply server/channels/app/file.go]: Fixed

**Comment by jespino** [comment]
For the QA team, this is not changing any behavior at all in the application, so it should work without problems. My understanding is, if automatic tests work properly, it should be fine. But I leave it to your judgement.

**Review by lindy65 — APPROVED** [review]
Thanks @jespino - E2E tests all passed so I'll approve this PR

**Pull request merged**